### PR TITLE
Implement generate-envelope workflow

### PR DESCRIPTION
**Pull request by qmuntal**
This PR implements the `generate-envelope` workflow by reusing most of the `PluginSigner` logic.

### CODE REVIEW

File: notation.go [hunk 1]
```diff
@@ -10,6 +10,9 @@ import (
 	"github.com/opencontainers/go-digest"
 )
 
+// MediaTypePayload describes the media type of the descriptor.
+const MediaTypePayload = "application/vnd.cncf.notary.payload.v1+json"
```
Comment (gokarnm) [inline notation.go:14]: This is the media type of a notary defined "wrapped" payload that currently contains only the descriptor.

**Review by gokarnm — APPROVED** [review]

File: signature/jws/plugin.go [hunk 2]
```diff
@@ -172,7 +179,104 @@ func (s *PluginSigner) mergeConfig(config map[string]string) map[string]string {
...
+	if err != nil {
+		return nil, err
+	}
+	// Execute plugin sign command.
+	req := &plugin.GenerateEnvelopeRequest{
+		ContractVersion:       plugin.ContractVersion,
+		KeyID:                 s.KeyID,
+		Payload:               rawDesc,
+		SignatureEnvelopeType: notation.MediaTypeJWSEnvelope,
+		PayloadType:           notation.MediaTypePayload,
```
[the remark this answers is not in the document]
Comment (qmuntal) [reply signature/jws/plugin.go]: Done!

**Comment by gokarnm** [comment]
@shizhMSFT this PR needs one more review, could you review?

**Comment by gokarnm** [comment]
@shizhMSFT this PR needs one more review, could you review?

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Comment by qmuntal** [comment]
As a small follow for https://github.com/notaryproject/notation-go/pull/46#discussion_r880094252, I've removed all uses of `signingMethodFromKey` in https://github.com/notaryproject/notation-go/pull/47/commits/3e47fc8ffae76c8b44e509e874ba00f11cb4d214, with the nice benefit being less dependent on `golang-jwt/jwt`.

**Review by SteveLasker — APPROVED** [review]
LGTM

**Pull request merged**